src/command/config.rs

```diff
@@ -0,0 +1,192 @@
+use std::fmt::{self, Display};
+use std::str::FromStr;
+
+use docopt::Docopt;
+use serde::Deserialize;
+
+use notion_core::session::Session;
+use notion_fail::{FailExt, Fallible};
+
+use Notion;
+use command::{Command, CommandName, Help};
+
+use CliParseError;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Args {
+    arg_command: SubcommandName,
+    arg_args: Option<Vec<String>>,
+}
+
+#[derive(Debug, Deserialize, Clone, Copy)]
+pub(crate) enum SubcommandName {
+    Get,
+    Set,
+    Delete,
+    List,
+    Edit,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Key {
+    arg_key: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct KeyValue {
+    arg_key: String,
+    arg_value: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Nullary;
+
+impl Display for SubcommandName {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        write!(
+            fmt,
+            "{}",
+            match *self {
+                SubcommandName::Get => "get",
+                SubcommandName::Set => "set",
+                SubcommandName::Delete => "delete",
+                SubcommandName::List => "list",
+                SubcommandName::Edit => "edit",
+            }
+        )
+    }
+}
+
+impl FromStr for SubcommandName {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "get" => SubcommandName::Get,
+            "set" => SubcommandName::Set,
+            "delete" => SubcommandName::Delete,
+            "list" => SubcommandName::List,
+            "edit" => SubcommandName::Edit,
+            _ => {
+                throw!(());
+            }
+        })
+    }
+}
+
+pub(crate) enum Config {
+    Help,
+    Subcommand(Subcommand),
+}
+
+pub(crate) enum Subcommand {
+    Get {
+        // Not yet implemented.
+        #[allow(dead_code)]
+        key: String,
+    },
+    Set {
+        // Not yet implemented.
+        #[allow(dead_code)]
+        key: String,
+
+        // Not yet implemented.
+        #[allow(dead_code)]
+        value: String,
+    },
+    Delete {
+        // Not yet implemented.
+        #[allow(dead_code)]
+        key: String,
+    },
+    List,
+    Edit,
+}
+
+fn parse_subcommand<'de, T: Deserialize<'de>>(
+    subcommand: &str,
+    usage: &str,
+    mut args: Vec<String>,
+) -> Fallible<T> {
+    let mut argv = vec![
+        "notion".to_string(),
+        "config".to_string(),
+        subcommand.to_string(),
+    ];
+    argv.append(&mut args);
+    let usage = format!("Usage: notion config {} {}", subcommand, usage);
+    Docopt::new(&usage[..])
+        .and_then(|d| d.argv(argv).deserialize())
+        .map_err(|err| err.with_context(CliParseError::from_docopt))
+}
+
+impl Command for Config {
+    type Args = Args;
+
+    const USAGE: &'static str = "
+Get or set configuration values
+
+Usage:
+    notion config <command> [<args> ...]
+    notion config -h | --help
```

Comment:
I think I had more the npm mental model, where `notion config` is more of a low-level command you don't use in day-to-day workflows.

It would probably be helpful for the usage info to separate out normal workflows from administrative tasks, to make it easier to learn the tool and to cut down on cognitive load. (Not sure if this is possible with docopt; at some point after MVP I want to look into replacing it with structopt or clap.)